ports/esp32/modesp32.c

```diff
@@ -212,6 +212,39 @@ static mp_obj_t esp32_idf_heap_info(const mp_obj_t cap_in) {
 }
 static MP_DEFINE_CONST_FUN_OBJ_1(esp32_idf_heap_info_obj, esp32_idf_heap_info);
 
+#if CONFIG_FREERTOS_USE_TRACE_FACILITY
+static mp_obj_t esp32_idf_task_stats(void) {
+    const size_t task_count_max = uxTaskGetNumberOfTasks();
+    TaskStatus_t *task_array = m_new(TaskStatus_t, task_count_max);
+    uint32_t total_time;
+    const size_t task_count = uxTaskGetSystemState(task_array, task_count_max, &total_time);
+
+    mp_obj_t task_list = mp_obj_new_list(0, 0);
```

Comment:
The second argument here should be NULL.

And the first argument could be `task_count`.  Then instead of `mp_obj_list_append` do:
```c
task_list->items[i] = task;
```
(Need to cast `task_list` to `mp_obj_list_t *` though... maybe that's a little less readable, up to you!)